openmls/src/binary_tree/array_representation/tree.rs

```diff
@@ -169,25 +182,26 @@ impl<T: Clone + Debug> ABinaryTree<T> {
     ///
     /// Returns an error if the leaf is outside of the tree.
     pub(crate) fn leaf(&self, leaf_index: LeafIndex) -> Result<&T, ABinaryTreeError> {
-        let node_index = usize::try_from(to_node_index(leaf_index))
-            .map_err(|_| ABinaryTreeError::ArchitectureError)?;
+        let node_index = usize::try_from(to_node_index(leaf_index)).map_err(|_| {
+            LibraryError::custom("ABinaryTree::merge_diff(): Architecture not supported")
+        })?;
         self.nodes
             .get(node_index)
             .ok_or(ABinaryTreeError::OutOfBounds)
     }
 }
 
-implement_error! {
-    pub enum ABinaryTreeError {
-        OutOfRange = "Adding nodes exceeds the maximum possible size of the tree.",
-        NotEnoughNodes = "Not enough nodes to remove.",
-        InvalidNumberOfNodes = "The given number of nodes does not allow the creation of a full, left-balanced binary tree.",
-        OutOfBounds = "The given index is outside of the tree.",
-        AddressCollision = "Found two nodes with the same address.",
-        InvalidNode = "Can't add the default node to the tree.",
-        NodeNotFound = "Can't find the node with the given address in the tree.",
-        LibraryError = "An unrecoverable error has occurred due to a bug in the implementation.",
-        ArchitectureError = "Architectures below 32bit are not suppored by OpenMLS.",
-        ABinaryTreeDiffError = "An error occurred while handling a diff.",
-    }
+/// Binary Tree error
```

Comment:
The comment is "Binary Tree error" but the enum is `ABinaryTreeError`. The comment doesn't help to understand what this is and only confuses.